Source/engine/points_in_rectangle_range.hpp

```diff
@@ -0,0 +1,195 @@
+#pragma once
+
+#include "point.hpp"
+#include "rectangle.hpp"
+
+namespace devilution {
+
+class PointsInRectangleRange {
+public:
+	using const_iterator = class PointsInRectangleIterator {
+	public:
+		using iterator_category = std::random_access_iterator_tag;
+		using difference_type = int;
+		using value_type = Point;
+		using pointer = const value_type *;
+		using reference = const value_type &;
+
+		PointsInRectangleIterator(Rectangle region, int index = 0)
+		    : region(region)
+		    , index(index)
+		{
+		}
+
+		value_type operator*() const
+		{
+			// Default to row-major iteration e.g. {0, 0}, {1, 0}, {2, 0}, {0, 1}, {1, 1}, ...
+			return region.position + Displacement { index % region.size.width, index / region.size.width };
```

Comment:
Not important for small rectangles, but if we start using this in larger rectangles, it'd be nice to avoid a modulo and an integer division on every iteration (can simply keep 2 indices instead of 1).